src/abstractdataframe/selection.jl

```diff
@@ -32,12 +32,7 @@ struct ByRow{T}
     fun::T
 end
 
-(f::ByRow)(col::AbstractVector) = (f.fun).(col)
-
-function (f::ByRow)(cols::NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}})
-    rowiterator = Tables.rows(cols)
-    map(f.fun, Tables.namedtupleiterator(eltype(rowiterator), rowiterator))
-end
+(f::ByRow)(cols::AbstractVector...) = [f.fun(row...) for row in zip(cols...)]
```

Comment:
Would it we useful to add `@inbounds` here? Ideally iteration would automatically enable it, but I never know whether it works (especially with `zip`). Another solution would be to use `eachindex` explicitly.

Reply:
I do not think `@inbounds` would change anything here (we do not do any indexing).

We could write something like (here `@inbounds` would be needed):
```
@inbounds [f.fun(ntuple(j -> cols[j][i], length(cols))...) for i in eachindex(cols...)]
```
or
```
@inbounds [f.fun(getindex.(cols, i)...) for i in eachindex(cols...)]
```

And the benchmarks show that `zip` is a bit faster for short vectors (like up to 10,000) and the other two are very comparable and are faster for long vectors. So I am not sure what is best.

(if we want to squeeze out maximum performance here do you know whom to ask for advice?)

Reply:
OK. Have you just checked that SIMD is enabled in both cases? Otherwise `@inbounds` may not make any difference.

(Not sure who we can ask. Maybe on #arrays if needed, but a simple benchmark may be enough.)

Reply:
I benchmarked if with `@inbounds` and without it and there is no difference (in the `zip` version). I used `+` as an aggregation function in the tests.

Reply:
Yeah, but can you see SIMD instructions if you use indexing instead of `zip`?

Reply:
Also, doesn't `@inbounds` need to be inside the comprehension?

Reply:
The simplest:
```
(f::ByRow)(cols::AbstractVector...) = f.fun.(cols...)
```
is the only one that does seem to generate SIMD for operation like `+`. And it is fastest for large vectors. But slowest for small vectors, so :confused:.

But maybe we should broadcast in the end - it is the cleanest solution (the biggest drawback is that it will have a tad bigger compilation time, but I think it is OK).

Reply:
OK, let's do that then. We should optimize for large data sets.

Just to be sure: `[f.fun(ntuple(j -> @inbounds(cols[j][i]), length(cols))...) for i in eachindex(cols...)]` doesn't generate SIMD instructions either?

Reply:
Well - in the morning I checked it on Windows and it seemed it did, now I check in on Linux and I do not see them. Anyway - given the @quinnj comment in #2123 we should not use `eachindex` anyway so it is not an option. Therefore I will go for plain broadcasting (which seems most elegant anyway).

Reply:
Well with `for i in eachindex(cols...)`, you don't index into the object returned by `eachindex` so #2123 shouldn't be a problem. But I'm fine with using broadcast here. Maybe leave a comment to explain why we use that instead of a comprehension with `zip`.

Reply:
Ah - right.

I do not think we need a comment as using broadcast is a natural thing to do.
I used a comprehension with `zip` because I wanted to save some compilation time.